### DISCUSSION TITLE
Home Mini issues this morning.

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#445] [General]
Has anyone else's Home Mini not received any date since 7am, my Home Mini LED is white, but this integration and Octopus's App has not received any data since 7am this morning.

**Comment by CambsWolfie** [comment]
It looks like I’ve not received any data either. I’m not at home so can’t check the lightPOn 7 Oct 2023, at 11:40, Andrew Blackburn ***@***.***> wrote:﻿
Has anyone else's Home Mini not received any date since 7am, my Home Mini LED is white, but this integration and Octopus's App has not received any data since 7am this morning.

**Comment by andyblac** [comment]
ok, thanks, So it sounds like it a Octopus Server issue, another reason they really need to sort out local api.

**Reply by deltamelter** [reply]
I had assumed that the integration was talking directly to the mini, is the data fed back up to octopus and published via API there?
What does the API look like to fetch live demand.  Just curious now that its not working :)

**Reply by andyblac** [reply]
as far as I know, yes, the data is fed to OE, then the API calls on that data, as for what it looks like you'll have to ask @BottlecapDave

**Reply by BottlecapDave** [reply]
Alas the mini sends the data to the OE servers, and the integration uses the same APIs as the app to pull the data down. So something isn't working on OE end if the app is not receiving the data as well. However, based on the CEO's tweets it sounds like they're working on a local solution for the Mini (probably because we keep flooding their severs).

**Reply by deltamelter** [reply]
Hi @BottlecapDave, which specific API command fetches live "demand" from OE? I only know the REST call for consumption which is previous day 30 min intervals.  is it using graphql query?

**Reply by BottlecapDave** [reply]
Everything to do with the mini is done via their GraphQL API and is not accessible via the REST endpoint. The query can be found [here](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/custom_components/octopus_energy/api_client/__init__.py#L119) (which has been escaped for python)

**Comment by andyblac** [comment]
@BottlecapDave will this missing data be picked up ?, it seem the Mini. is still not getting any data, what will happen to the energy dashbaord, as I dont use `previous` i use `current`

**Reply by BottlecapDave** [reply]
Assuming it's fixed today, the data should be picked up, but it will be a big spike during the period it's picked up. This is a limitation of HA.

**Reply by andyblac** [reply]
and if it not ?, is there a way inject the missing data when it available ?

**Reply by BottlecapDave** [reply]
There isn't I'm afraid unless you want to go sql diving and inject the data yourself. HA doesn't like you altering historic data (it gets confused when trying to reconcile the state of the sensor), which is why the previous consumption breakdowns are in a separate statistic sensor.

**Comment by andyblac** [comment]
ok thanks.

**Comment by anothermort** [comment]
Have been getting electricity all day as normal. Just no gas.
Hope fully it gets sorted for local data that would be awesome.

**Reply by pcb1962** [reply]
If you want local data now you can use a glow device - https://shop.glowmarkt.com/

**Reply by andyblac** [reply]
unfortunately for me that wont work, as I have tried 4 types of IHD's (including GEO) , none will connect to my meters.

**Comment by andyblac** [comment]
@BottlecapDave 

just got some data through, but there was no big spike
<img width="1257" alt="Screenshot 2023-10-07 at 17 13 02" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/d62535c6-bda6-4218-9ad9-f703389da057">


Scrap that, I see it now, it was after 17:00 so it should come through in the 17:00 - 18:00 block
<img width="1545" alt="Screenshot 2023-10-07 at 17 15 56" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/12136c07-59ba-4a81-9c9e-1555e6b2c845">

**Reply by davidmrowe** [reply]
I wish I'd found this thread earlier rather than scratching my head for ages! Mine is now showing data in HA although some crazy spikes - see the comparison with yesterday - hopefully it'll settle down.

I'm not seeing any data in the app on my phone though. The mini is showing a white light which apparently means it's happy.  I'll just wait it out and see what happens.

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/16492689/06c39537-7011-4059-9ba0-cd10a9721d70)

**Comment by deltamelter** [comment]
Seems delayed and showing at the wrong time.  Just boiled a kettle at 19:55 and only showed up on the graph 15-20 mins later.
